File: benchmark/py/06_string.py
# 文字列処理 (反復生成・置換・検索) --- 06_string.nako3 と同じアルゴリズム
total_len = 0
work = ""

for _ in range(3000):
    work = work + "いろはにほへと"
    if len(work) > 200:
        work = work.replace("いろは", "なでしこ")
        # なでしこの「文字検索」は1から数えた位置。見つからなければ0
        pos = work.find("なでしこ", 0) + 1
        total_len = total_len + pos
        work = work[0:100]
    total_len = total_len + len(work)

print("string(3000) = total_len:{}, final_len:{}".format(total_len, len(work)))

File: benchmark/js/06_string.js
// 文字列処理 (反復生成・置換・検索) --- 06_string.nako3 と同じアルゴリズム
let totalLen = 0
let work = ''

for (let i = 0; i < 3000; i++) {
  work = work + 'いろはにほへと'
  if (work.length > 200) {
    work = work.split('いろは').join('なでしこ')
    // なでしこの「文字検索」は1から数えた位置。見つからなければ0
    const pos = work.indexOf('なでしこ', 0) + 1
    totalLen = totalLen + pos
    work = work.substring(0, 100)
  }
  totalLen = totalLen + work.length
}

console.log(`string(3000) = total_len:${totalLen}, final_len:${work.length}`)
